typescript/infra/config/environments/mainnet3/chains.ts

```diff
@@ -1,17 +1,30 @@
-import {
-  ChainMap,
-  ChainMetadata,
-  Mainnets,
-  chainMetadata,
-} from '@hyperlane-xyz/sdk';
+import { ChainMap, ChainMetadata, ChainName } from '@hyperlane-xyz/sdk';
 
 import { getChainMetadatas } from '../../../src/config/chain.js';
+import { getChain } from '../../registry.js';
 
-// The `Mainnets` from the SDK are all supported chains for the mainnet3 environment.
 // These chains may be any protocol type.
-export const supportedChainNames = Mainnets;
+export const supportedChainNames = [
```

Comment:
Listing chains explicitly here like we already do for testnet, otherwise the tests break due to extra chains in the registry that aren't in the agent config

Reply:
This works for me and is probably preferable anyways. I imagine that the registry will eventually have e.g. PI mainnets in there that we don't want to run any validators / relayers on, so explicitly calling out which mainnets we want to run infra for makes sense